shared/src/main/scala/com/nitro/nmesos/config/YamlParserHelper.scala

```diff
@@ -100,7 +95,7 @@ object YamlParserHelper {
   private def smartMerge(initialFields: Map[YamlValue, YamlValue], element: YamlObject): YamlObject = {
     val fields = element.fields.foldLeft(initialFields) {
       case (mergedFields, (key, value)) =>
-        if (!mergedFields.contains(key)) {
+        if (!mergedFields.contains(key) || mergedFields(key) == YamlNull) {
```

Comment:
This is the bug-fix. I am not only checking for mergedFields not having the key, but also for the key to be `YamlNull`.